backend/core/src/secondary-adapters/postgres/history/history-repository-adapter.ts

```diff
@@ -11,19 +13,81 @@ export class HistoryRepositoryAdapter implements HistoryRepository {
     this.prismaClient = prismaClient;
   }
 
-  getAllUserHistory(userId: string): Promise<History[]> {
-    return this.prismaClient.history.findMany({
+  async getAllUserHistoryLength(userId: string): Promise<number> {
+    return this.prismaClient.history.count({
       where: {
         userId,
       },
     });
   }
 
-  async createHistoryItem(historyRequestDto: HistoryRequestDto): Promise<HistoryResponseDto> {
-    const history = await this.prismaClient.history.create({
-      data: { ...historyRequestDto },
+  async getUserHistory(userId: string, take: number, skip: number, lastPage: number): Promise<HistoryResponseDto> {
+    const history = await this.prismaClient.history.findMany({
+      take,
+      skip,
+      where: {
+        userId,
+      },
+      include: {
+        video: {
+          include: {
+            channel: {
+              select: {
+                id: true,
+                name: true,
+                avatar: true,
+              },
+            },
+          },
+        },
+      },
+      orderBy: {
+        updatedAt: 'desc',
+      },
     });
+    const list: HistoryListType[] = history.map((historyRecord) => trimHistory(historyRecord));
 
-    return history;
+    return {
+      list,
+      lastPage,
+      currentPage: skip / 10 + 1,
```

Comment:
if this 10 is related to the previous 10s (from` backend/core/src/core/history/application/history-service.ts`), be sure to use a constant here as well, this file and the previous file should reference the same constant

Reply:
done

Reply:
done